auth/api/iam/client.go

```diff
@@ -80,3 +81,39 @@ func (hb HTTPClient) OAuthAuthorizationServerMetadata(ctx context.Context, webDI
 
 	return &metadata, nil
 }
+
+// PresentationDefinition retrieves the presentation definition for the given web DID and scope(s).
+// We pass the endpoint url for the presentation definition endpoint because we already retrieved the metadata in a previous step.
+// The scopes are evaluated as raw query params and encoded if needed.
+func (hb HTTPClient) PresentationDefinition(ctx context.Context, definitionEndpoint string, scopes []string) ([]PresentationDefinition, error) {
+	presentationDefinitionURL, err := url.Parse(definitionEndpoint)
```

Comment:
if you want a `url.URL`, you can just ask for it (string -> url.URL)?

Reply:
You get it as a string from the metadata, so it would move the parsing to the caller. That would duplicate parsing code for every call.

Reply:
Putting ur.URL in the metadata will require custom marshalling/unmarshalling for the metadata struct.